src/app/components/LocationMap/MapwithMarker.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useRef, useEffect, useState } from 'react';
+import mapboxgl from 'mapbox-gl';
+import type { Map } from 'mapbox-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import styled from 'styled-components';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
+
+type MapWithMarkerProps = {
+  displayArea: string;
+};
+
+export default function MapWithMarker({
+  displayArea,
+}: MapWithMarkerProps): JSX.Element {
+  if (typeof import.meta.env.VITE_MAPBOX_ACCESS_KEY === 'string') {
+    mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_KEY;
+  } else {
+    throw new Error('no KEY provided');
+  }
```

Comment:
It's uncommon to check if environment variables are set during render cycle.
Instead, you could move this logic to outside `MapWithMarker`.